mininet_spinup.py

```diff
@@ -0,0 +1,194 @@
+        containers, referenced by their names
+    '''
+
+    def __init__(self):
+        '''
+        Creates the DockerClient needed to communicate with the Docker daemon.
+        Development and production Docker images are initially None until built
+        within this instance.
+        '''
+
+        self.client = docker.from_env()
+        self.dev_image = None
+        self.prod_image = None
+        self.running_containers = dict()
+
+    def run_mininet_container(self, topo_file, dev=True):
+        '''
+        Runs the specified Docker image as a container and binds the provided
+        file for Toynet to parse its initial topology from.
+
+        topo_file: String - the absolute path of the Toynet topology file to
+                            map into the container
+        dev: bool - True if we want to run the developer image, False for
+                    production
+
+        returns: String - the name of the container running, empty string if no
+                          image is available
+        '''
+
+        vol = dict()
+        vol[topo_file] = {'bind': '/root/toynet-mininet/topo.xml', 'mode':'ro'}
+        vol['/lib/modules'] = {'bind': '/lib/modules', 'mode':'ro'}
+        image_name = ''
+
+        if dev and self.dev_image is not None:
+            image_name = self.dev_image[0]
+        elif not dev and self.prod_image is not None:
+            image_name = self.prod_image[0]
+        else:
+            return image_name
+
+        new_container = self.client.containers.run(image_name, privileged=True,
+                                                   remove=True, detach=True, volumes=vol)
+        self.running_containers[new_container.name] = new_container
+
+        return new_container.name
+
+    def build_mininet_container(self, dev=True):
+        '''
+        Builds the Toynet container from the local Dockerfile or dev.Dockerfile.
+
+        dev: bool - True indicates to build the dev.Dockerfile image, False
+                    indicates to build the production Dockerfile image
+
+        returns: void
+        '''
+
+        if dev:
+            self.dev_image = self.client.images.build(path='.', dockerfile='dev.Dockerfile',
+                                                      rm=True, tag='toynet-dev')
+        else:
+            self.prod_image = self.client.images.build(path='./', rm=True, tag='toynet')
+
+    def kill_container(self, container_name):
+        '''
+        Kills the specified Docker container if running.
+
+        conainer_name: String - the name of the container to kill
+
+        returns: True if the container was killed, False otherwise
+        '''
+
+        res = True
+
+        try:
+            self.running_containers[container_name].kill()
+            del self.running_containers[container_name]
+        except (KeyError, docker.errors.APIError) as error:
+            print(error)
+            res = False
```

Comment:
If we don't catch this exception, I believe we would return nothing, so it may make more sense to just return `True` at the end and not worry about `False`